react/week3/src/todoItem.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+
+const ToDoItem = (props) => {
+
+  const selectCheckBox = () => {
+    console.log('selectCheckBox');
+    console.log(props.shift.completed)
+    props.shift.completed = !props.shift.completed;
+    props.setShifts(prev => prev.filter((item) => item))
+  }
+
+  const editTodo = (value) => {
+    console.log('editTodo');
+    console.log(value);
+    props.shift.description = value;
+    props.setShifts(prev => prev.filter((item) => item))
```

Comment:
This doesn't really do anything, I assume you added it cause you wanted your component to re-render, a bit hacky :)
The problem though is that you are only changing the completed property here, but not on your shifts. What you want to do, is to define a new description state variable here, which will change if you edit the textbox. And then when the 'update' button is clicked you can update the whole shifts array by:
```
setTodos((prev) => {
        const newTodos = prev;
        return newTodos.map((item) => {
          if (item.id === todo.id) {
            return { ...item, description: editedDescription };
          }
          return item;
        });
      });
```